Tests/ApolloTests/StarWars/API.swift

```diff
@@ -54,59 +62,68 @@ public final class CreateReviewForEpisodeMutation: GraphQLMutation {
   public struct Data: GraphQLMappable {
     public let createReview: CreateReview?
 
-    public init(reader: GraphQLResultReader) throws {
-      createReview = try reader.optionalValue(for: Field(responseName: "createReview", arguments: ["episode": reader.variables["episode"], "review": reader.variables["review"]]))
+    public init(values: [Any?]) {
+      createReview = values[0] as! CreateReview?
     }
 
     public struct CreateReview: GraphQLMappable {
       public let __typename: String
       public let stars: Int
       public let commentary: String?
 
-      public init(reader: GraphQLResultReader) throws {
-        __typename = try reader.value(for: Field(responseName: "__typename"))
-        stars = try reader.value(for: Field(responseName: "stars"))
-        commentary = try reader.optionalValue(for: Field(responseName: "commentary"))
+      public init(values: [Any?]) {
+        __typename = values[0] as! String
```

Comment:
The difference is that before, code like `reader.value(for: Field(responseName: "stars"))` would be driving the parsing. There was no separate runtime representation of the query, so accessing a value would both resolve the value (from response or cache) and convert it to the right type.

The new execution model uses a runtime representation to basically implement the [GraphQL execution model from the spec](https://facebook.github.io/graphql/#sec-Execution), with some changes. It merges selections sets and collects fields for resolving (so we only resolve once), but it still keeps track of the original selections so we can instantiate the right types.

So for a query like: 

```graphql
query HeroAndFriendsNames {
  hero {
    name
    ...HeroDetails
    friends {
      name
    }
}

fragment HeroDetails on Character {
  id
  friends {
    appearsIn
  }
}
```

`friends` would only be resolved once, and we would then resolve both `name` and `appearsIn` for all friends. But we would still instantiate a separate `HeroAndFriendsNames.Data.Hero.Friend` and `HeroDetails.Friend` from that same information.

The `as!` is as forced cast, because the execution model is responsible for instantiating/converting the right types.